xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
Just for clarification: I didn't make this patch because I saw an actual error happen. But someone not familiar with the code may expects `IsRunning()` (and also `ThreadId()` btw.) to always return the actual state of the thread at any time.